src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ServiceMessageBus.cs

```diff
@@ -2,24 +2,74 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Infrastructure;
 using Microsoft.AspNet.SignalR.Messaging;
+using Microsoft.Azure.SignalR.Protocol;
 
 namespace Microsoft.Azure.SignalR.AspNet
 {
-    /// <summary>
-    /// TODO: This class is responsible for transform SignalR Messages to ServiceMessages and send them to the service runtime
-    /// </summary>
     internal class ServiceMessageBus : MessageBus
     {
+        private readonly IMessageParser _parser;
+        private readonly IServiceConnectionManager _serviceConnectionManager;
+        private readonly IAckHandler _ackHandler;
+
         public ServiceMessageBus(IDependencyResolver resolver) : base(resolver)
         {
+            _serviceConnectionManager = resolver.Resolve<IServiceConnectionManager>() ?? throw new ArgumentNullException(nameof(IServiceConnectionManager));
```

Comment:
This will be problematic but we'll see the reports of what doesn't work with custom DI containers.

Reply:
There's some weirdness here around passing in the resolver to the constructor. It probably won't be the right one if you override the `IDependencyResolver`.

Reply:
Add a TODO for this